lib/config/presets/internal/config.ts

```diff
@@ -15,6 +15,20 @@ export const presets: Record<string, Preset> = {
       'workarounds:all',
     ],
   },
+  'best-practice': {
+    description: 'Preset with what renovate considers best practice.',
+    extends: [
+      'config:base',
+      'docker:pinDigests',
+      'helpers:pinGitHubActionDigests',
+    ],
+    packageRules: [
+      {
+        matchDepTypes: ['devDependencies'],
+        rangeStrategy: 'pin',
+      },
+    ],
```

Comment:
Are there any more best practices that the Renovate maintainers want to add here?

Reply:
We should check our own presets to see if any worth adding

Reply:
I found some things in our [`default.json` in the `renovatebot/.github` repository](https://github.com/renovatebot/.github/blob/main/default.json) to think about.

### `docker:enableMajor`:

This one is a bit odd for me. I would expect extending from `config:base` would give you `major` Docker image upgrades out-of-the-box. There's probably a reason we don't enable `major` Docker images by default for `config:base`?

I think as a best practice, people should use the latest `major` version of their Docker images, or at least be aware that there's a new `major` version.

```json
{
  "packageRules": [
    {
      "enabled": true,
      "matchDatasources": [
        "docker"
      ],
      "matchUpdateTypes": [
        "major"
      ]
    }
  ]
}
```

### `group:linters`:

Grouping linters is usually safe, but I guess most people want separate updates?

```json
{
  "packageRules": [
    {
      "extends": "packages:linters",
      "groupName": "linters"
    }
  ]
}
```

And this is what `packages:linters` does:

```json
{
  "extends": [
    "packages:emberTemplateLint",
    "packages:eslint",
    "packages:stylelint",
    "packages:tslint"
  ],
  "matchPackageNames": [
    "prettier",
    "remark-lint",
    "standard"
  ]
}
```

### `configMigration=true`

After this issue is done:

- #16359

We should include `configMigration=true` in our best practices preset once it's ready to go. 😄

### Update version variables in shell scripts/GitHub Workflows:

I like the idea of this config, we'd need to check if this generalizes to other configs though:

```json
    {
      "description": "Update _VERSION variables in shell scripts",
      "fileMatch": ["\\.sh$"],
      "matchStrings": [
        "# renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>.+?)(?: (?:packageName|lookupName)=(?<packageName>.+?))?(?: versioning=(?<versioning>[a-z-]+?))?\\s[A-Z_]+?_VERSION=(?<currentValue>.+?)\\s"
      ],
      "versioningTemplate": "{{#if versioning}}{{versioning}}{{else}}semver{{/if}}"
    },
    {
      "description": "Update `version:` and `_VERSION:` variables in github workflows",
      "fileMatch": ["^\\.github/workflows/[^/]+\\.ya?ml$"],
      "matchStrings": [
        "\\s+version: (?<currentValue>.+?) # renovate: datasource=(?<datasource>.+?) depName=(?<depName>.+?)(?: (?:packageName|lookupName)=(?<packageName>.+?))?(?: versioning=(?<versioning>.+?))?\\s",
        "\\s*[A-Z_]+?_VERSION: (?<currentValue>.+?) # renovate: datasource=(?<datasource>.+?) depName=(?<depName>.+?)(?: (?:packageName|lookupName)=(?<packageName>.+?))?(?: versioning=(?<versioning>.+?))?\\s"
      ]
    }
```

### Group Prettier packages

This looks nice to have as well.

By the way, this looks like a good candidate for a new preset?

```json
    {
      "description": "Group prettier packages",
      "matchPackageNames": ["prettier"],
      "matchPackagePatterns": ["^@prettier/", "^prettier-plugin-"],
      "groupName": "prettier packages"
    },
```

### Dashboard Approval for `major` updates?

For me, I'd rather get the `major` updates automatically, but maybe you guys prefer the approval first flow?

```json
    {
      "description": "Require dashboard approval for major updates except Renovate",
      "excludePackagePatterns": [
        "^renovate(bot)?($|/)",
        "^(?:ghcr.io/)?containerbase/base$",
        "^@renovatebot/"
      ],
      "matchUpdateTypes": ["major"],
      "dependencyDashboardApproval": true
    },
```

Reply:
docker:enableMajor will probably have something to do about the fact that versions in docker images don't always follow semantic versioning standards?

Reply:
Is grouping something suited for a best practice preset?
Seems like something that should be decided by maintainers?
Is there a way to disable the grouping so users of the preset are not forced to copy paste the preset and remove the grouping to disable it?

if we add `configMigration=true` here then I assume it not added to `config:base`? if the plan is to add it to `config:base` then this would be inherited.

Reply:
> `docker:enableMajor` will probably have something to do about the fact that versions in docker images don't always follow semantic versioning standards?

I don't know. 😄 


> Is grouping something suited for a best practice preset?
> Seems like something that should be decided by maintainers?
> Is there a way to disable the grouping so users of the preset are not forced to copy paste the preset and remove the grouping to disable it?

I'm not quite sure what you mean here. The `config:base` preset already groups certain packages and monorepos together. Do you mean that our new `config:best-practice(s)` preset should _not_ add any more groupings?

> if we add `configMigration=true` here then I assume it not added to `config:base`? if the plan is to add it to `config:base` then this would be inherited.

We have an issue that tracks setting config migration to `true` as the default behavior. I don't know _how_ we're going to make that the default behavior though: 🙈 

- https://github.com/renovatebot/renovate/issues/16359

Reply:
@rarkins can you review this thread, and let us know if there's anything you want to add to our opinionated `best-practices` preset? If you're happy with things as they are, let us know also. 😄

Reply:
- `docker:enableMajor` is no longer needed
- Skip `group:linters`
- Include `configMigration=true`
- We should probably convert our regex managers into presets but not sure there's any point including them by default
- I think dashboard approval for major updates is generally a good idea, but not certain whether to put it in config:recommended

Reply:
We should check if any of our org's regex managers are _not_ in these presets

Reply:
> We should check if any of our org's regex managers are _not_ in these presets

I checked, but a maintainer will need to review my work. There are four repositories that have things that may be useful to others. See issue:

- #22025